src/Guild.js

```diff
@@ -25,6 +25,35 @@ class Guild {
     autoBind(this);
   }
 
+  guildRoleCreate (roleId) {
+    this.roles[this.roles.name] = roleId;
+  }
+
+  guildGetKeyByValue (value) {
+    return Object.keys(this.roles).find(key => this.roles[key] === value);
+  }
+
+  guildUpdateRoleName (roleId) {
+    const oldKey = this.guildGetKeyByValue(roles.id);
+
+    //role for roleId does not exist. nothing to update
+    if(!this.isRole(oldKey)) {
+      return;
+    }
+
+    if(oldKey !== this.roles.name) {
```

Comment:
`this.roles.name` is getting the value assigned to the key `name`, not the name of the Role. You need to pass the Role name in.

If you follow the previous suggestion, you won't need this if statement. This is because if the Role name doesn't exist in `this.roles`, you know the name has changed.